scripts/zones/Bastok_Markets/npcs/Gulldago.lua

```diff
@@ -4,16 +4,116 @@
 -- Type: Tutorial NPC
 -- !pos -364.121 -11.034 -167.456 235
 -----------------------------------
+local ID = require("scripts/zones/Bastok_Markets/IDs")
+require("scripts/globals/status")
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
+require("scripts/globals/npc_util")
+-----------------------------------
+
 
-function onTrade(player,npc,trade)
-end;
+function onTrade(player, npc, trade)
+end
 
-function onTrigger(player,npc)
-    player:startEvent(535);
-end;
+function onTrigger(player, npc)
+    local tutorial = player:getCharVar("TutorialProgress")
+    if tutorial == 11 then
+        player:startEvent(535)
+    else
+        local isSkilled = false
+        local mLevel = player:getMainLvl()
+        local hasCrystal = player:hasKeyItem(tpz.ki.DEM_GATE_CRYSTAL)
+        if tutorial == 3 then
+            for i = 1, 12 do
+                if player:getSkillLevel(i) >= 5 then
+                    isSkilled = true
+                    break
+                end
+            end
+        end
+        if tutorial == 0 or tutorial == nil then
```

Comment:
`getCharVar` will always return 0 if the variable isn't in the DB, so you don't need to check if it's `nil` here.

(But good looking out for this sort of thing! The binding is just a little weird 😉 )